This patch explicitly disables everything in Luanti that was
written for the .apk file builds of Luanti on F-Droid and
Google Play that is believed to be unusable or unnecessary inside Termux, including:

- JNI (Java-Native Interface) method calls to, and entrypoints from, the "AndroidX" "Android Runtime" "app_process" execution environment
- ANativeWindow backend of Irrlicht/SDL windowing libraries, required by the SurfaceFlinger display server
- bug fixes believed to be only necessary while using above APIs
- there might be other things not listed

but without disabling anything that was written for the .apk file
builds of Luanti that seems, or is, useful or necessary inside Termux,
including:

- OpenGL ES support
- OpenSL ES support
- Touchscreen Buttons
- Reduced default graphics settings
- Preexisting Bionic-libc porting code like threading using calls to pthread_kill() in place of pthread_cancel()
- there might be other things not listed

For maintainers - how to bump this patch checklist, for avoiding bitrot:

- open the source code of the current Luanti release, and count the number of instances of the string 'ANDROID' in all-caps
- open the source code of the next Luanti release side-by-side, and count the number of instances of the string 'ANDROID' there
- match up all ~120 instances of the string 'ANDROID' that are untouched between the releases, and ignore them
- if any instances of the string have been added, removed, or heavily modified, consider each change one by one and decide whether
    to leave it enabled, or disable it
- use git blame (commit history) of the luanti repository to learn important context including commit messages and bug reports
    written by the Luanti upstream developers
- remember, in particular, that Termux:X11 usually behaves similarly to the way Xorg does on a Touchscreen Linux Laptop,
    and therefore, it is unlikely to trigger application bugs specific to SurfaceFlinger (ANativeWindow) or Wayland,
	but is likely to instead trigger application bugs that upstream tracks as specific to Xorg
- test the results by trying to activate gameplay codepaths that enter the blocks of code surrounding the areas where the new instances
    of 'ANDROID' strings were found

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,7 +93,7 @@ set(ENABLE_UPDATE_CHECKER (NOT ${DEVELOPMENT_BUILD}) CACHE BOOL
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
 
 # Load default options for Android
-if(ANDROID)
+if(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 	cmake_minimum_required(VERSION 3.20)
 	include(AndroidLibs)
 endif()
--- a/irr/src/CIrrDeviceSDL.cpp
+++ b/irr/src/CIrrDeviceSDL.cpp
@@ -348,7 +348,7 @@ CIrrDeviceSDL::CIrrDeviceSDL(const SIrrlichtCreationParameters &param) :
 		IsInBackground(false)
 {
 	if (++SDLDeviceInstances == 1) {
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 		// Blocking on pause causes problems with multiplayer.
 		// see <https://github.com/luanti-org/luanti/issues/10842>
 		SDL_SetHint(SDL_HINT_ANDROID_BLOCK_ON_PAUSE, "0");
@@ -864,7 +864,7 @@ bool CIrrDeviceSDL::run()
 			irrevent.MouseInput.Event = EMIE_MOUSE_MOVED; // value to be ignored
 
 			auto button = SDL_event.button.button;
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 			// Android likes to send the right mouse button as the back button.
 			// According to some web searches I did, this is probably
 			// vendor/device-specific.
--- a/irr/src/CIrrDeviceSDL.h
+++ b/irr/src/CIrrDeviceSDL.h
@@ -172,7 +172,7 @@ public:
 		//! Sets the new position of the cursor.
 		void setPosition(s32 x, s32 y) override
 		{
-#if !defined(__ANDROID__) && !defined(_IRR_EMSCRIPTEN_PLATFORM_)
+#if !defined(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK) && !defined(_IRR_EMSCRIPTEN_PLATFORM_)
 			// On Android, this somehow results in a camera jump when enabling
 			// relative mouse mode and it isn't supported anyway.
 			// For Emscripten, browsers don't allow moving the pointer at all.
--- a/irr/src/CMakeLists.txt
+++ b/irr/src/CMakeLists.txt
@@ -77,7 +77,7 @@ if(WIN32)
 	add_compile_definitions(_IRR_WINDOWS_ _IRR_WINDOWS_API_)
 elseif(APPLE)
 	add_compile_definitions(_IRR_OSX_PLATFORM_)
-elseif(ANDROID)
+elseif(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 	add_compile_definitions(_IRR_ANDROID_PLATFORM_)
 elseif(EMSCRIPTEN)
 	add_compile_definitions(_IRR_EMSCRIPTEN_PLATFORM_ _IRR_COMPILE_WITH_EGL_MANAGER_)
@@ -190,7 +190,7 @@ else()
 	if(NOT USE_SDL2_STATIC)
 		set(USE_SDL2_SHARED TRUE)
 	endif()
-	if(NOT ANDROID)
+	if(NOT DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 		find_package(SDL2 REQUIRED)
 	else()
 		# provided by AndroidLibs.cmake
@@ -518,7 +518,6 @@ target_link_libraries(IrrlichtMt PRIVATE
 	${EGL_LIBRARY}
 
 	# incl. transitive SDL2 dependencies for static linking
-	"$<$<PLATFORM_ID:Android>:-landroid -llog -lGLESv2 -lGLESv1_CM -lOpenSLES>"
 )
 
 if(WIN32)
--- a/lib/catch2/catch_amalgamated.hpp
+++ b/lib/catch2/catch_amalgamated.hpp
@@ -9044,7 +9044,7 @@ namespace Catch {
 #define CATCH_CONFIG_ANDROID_LOGWRITE_HPP_INCLUDED
 
 
-#if defined(__ANDROID__)
+#if defined(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 #    define CATCH_INTERNAL_CONFIG_ANDROID_LOGWRITE
 #endif
 
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -354,7 +354,7 @@ else()
 		list(APPEND PLATFORM_LIBS network)
 	endif()
 
-	if (ANDROID)
+	if (DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 		list(APPEND PLATFORM_LIBS android log)
 	endif()
 endif()
@@ -532,7 +532,7 @@ set(common_SRCS
 	${unittest_SRCS}
 )
 
-if(ANDROID)
+if(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 	list(APPEND common_SRCS porting_android.cpp)
 endif()
 
@@ -607,7 +607,6 @@ include_directories(SYSTEM
 	${LUA_BIT_INCLUDE_DIR}
 	# on Android, Luanti depends on SDL2 directly
 	# on other platforms, only IrrlichtMt depends on SDL2
-	"$<$<PLATFORM_ID:Android>:${SDL2_INCLUDE_DIRS}>"
 )
 
 if(USE_GETTEXT)
@@ -663,7 +662,7 @@ endif()
 
 if(BUILD_CLIENT)
 	# client target
-	if(ANDROID)
+	if(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 		add_library(${PROJECT_NAME} SHARED)
 	else()
 		add_executable(${PROJECT_NAME})
@@ -1103,7 +1102,7 @@ macro(CreateLegacyAlias _cmake_target _old_path _new_name)
 	install(PROGRAMS "${_old_path}" DESTINATION ${BINDIR})
 endmacro()
 
-if(BUILD_CLIENT AND NOT ANDROID)
+if(BUILD_CLIENT AND NOT DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 	install(TARGETS ${PROJECT_NAME}
 		RUNTIME DESTINATION ${BINDIR}
 		LIBRARY DESTINATION ${BINDIR}
@@ -1141,7 +1140,7 @@ if(BUILD_SERVER)
 	endif()
 endif()
 
-if (ANDROID)
+if (DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 	# Android does this manually in app/build.gradle -> prepareAssets
 	# for now!
 elseif (USE_GETTEXT)
--- a/src/client/game.cpp
+++ b/src/client/game.cpp
@@ -499,7 +499,7 @@ void Game::run()
 	const bool initial_window_maximized = !g_settings->getBool("fullscreen") &&
 			g_settings->getBool("window_maximized");
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	porting::setPlayingNowNotification(true);
 #endif
 
@@ -597,7 +597,7 @@ void Game::run()
 
 	framemarker.end();
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	porting::setPlayingNowNotification(false);
 #endif
 
@@ -1409,7 +1409,7 @@ void Game::processUserInput(f32 dtime)
 	// Input handler step() (used by the random input generator)
 	input->step(dtime);
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	if (!m_game_formspec.handleAndroidUIInput())
 		handleAndroidChatInput();
 #endif
@@ -1437,7 +1437,7 @@ void Game::processKeyInput()
 	} else if (wasKeyDown(KeyType::INVENTORY)) {
 		m_game_formspec.showPlayerInventory(nullptr);
 	} else if (input->cancelPressed()) {
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 		m_android_chat_open = false;
 #endif
 		if (!gui_chat_console->isOpenInhibited()) {
@@ -1604,7 +1604,7 @@ void Game::openConsole(float scale, const wchar_t *line)
 {
 	assert(scale > 0.0f && scale <= 1.0f);
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	if (!porting::hasPhysicalKeyboardAndroid()) {
 		porting::showTextInputDialog("", "", 2);
 		m_android_chat_open = true;
@@ -1617,12 +1617,12 @@ void Game::openConsole(float scale, const wchar_t *line)
 		gui_chat_console->setCloseOnEnter(true);
 		gui_chat_console->replaceAndAddToHistory(line);
 	}
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	} // else
 #endif
 }
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 void Game::handleAndroidChatInput()
 {
 	// It has to be a text input
--- a/src/client/game_formspec.cpp
+++ b/src/client/game_formspec.cpp
@@ -103,7 +103,7 @@ struct HardcodedPauseFormspecHandler : public TextDest
 
 		if (fields.find("btn_exit_os") != fields.end()) {
 			g_gamecallback->exitToOS();
-#ifndef __ANDROID__
+#ifndef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 			RenderingEngine::get_raw_device()->closeDevice();
 #endif
 			return;
@@ -404,7 +404,7 @@ void GameFormSpec::showPauseMenu()
 		// TRANSLATORS: Try to keep the translation short
 		<< strgettext("Settings") << "]";
 
-#ifndef __ANDROID__
+#ifndef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 #if USE_SOUND
 	os << "button[0," << (ypos += button_height + spacing) << ";3.5," << button_height << ";btn_sound;"
 		// TRANSLATORS: Pause menu button, try to keep the translation short
@@ -579,7 +579,7 @@ bool GameFormSpec::handleCallbacks()
 	return true;
 }
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 bool GameFormSpec::handleAndroidUIInput()
 {
 	// FIXME: m_formspec and this value are not in sync at all times.
--- a/src/client/game_formspec.h
+++ b/src/client/game_formspec.h
@@ -46,7 +46,7 @@ struct GameFormSpec
 	bool handleCallbacks();
 	void reset();
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	// Returns false if no formspec open
 	bool handleAndroidUIInput();
 #endif
--- a/src/client/hud.cpp
+++ b/src/client/hud.cpp
@@ -390,7 +390,7 @@ void Hud::drawLuaElements(const v3s16 &camera_offset, bool only_unhidable)
 				if (e->size.X > 0)
 					font_size *= e->size.X;
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 				// The text size on Android is not proportional with the actual scaling
 				// FIXME: why do we have such a weird unportable hack??
 				if (font_size > 3 && e->offset.X < -20)
--- a/src/client/renderingengine.cpp
+++ b/src/client/renderingengine.cpp
@@ -163,7 +163,7 @@ RenderingEngine::RenderingEngine(MyEventReceiver *receiver)
 
 	// Resolution selection
 	bool fullscreen = g_settings->getBool("fullscreen");
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	u16 screen_w = 0, screen_h = 0;
 	bool window_maximized = false;
 #else
@@ -328,7 +328,7 @@ void RenderingEngine::draw_load_screen(const std::wstring &text,
 				tsrc->getTexture("progress_bar_bg.png");
 
 		if (progress_img && progress_img_bg) {
-#ifndef __ANDROID__
+#ifndef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 			const core::dimension2d<u32> &img_size =
 					progress_img_bg->getSize();
 			float density = g_settings->getFloat("gui_scaling", 0.5f, 20.0f) *
@@ -444,7 +444,7 @@ void RenderingEngine::showErrorMessageBox(const std::string &message)
 float RenderingEngine::getDisplayDensity()
 {
 	float user_factor = g_settings->getFloat("display_density_factor", 0.5f, 5.0f);
-#ifndef __ANDROID__
+#ifndef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	float dpi = get_raw_device()->getDisplayDensity();
 	if (dpi == 0.0f)
 		dpi = 96.0f;
--- a/src/defaultsettings.cpp
+++ b/src/defaultsettings.cpp
@@ -417,7 +417,7 @@ void set_default_settings()
 	settings->setDefault("contentdb_enable_updates_indicator", "true");
 	settings->setDefault("contentdb_max_concurrent_downloads", "3");
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	settings->setDefault("contentdb_flag_blacklist", "nonfree, android_default");
 #else
 	settings->setDefault("contentdb_flag_blacklist", "nonfree, desktop_default");
@@ -561,9 +561,6 @@ void set_default_settings()
 	settings->setDefault("clickable_chat_weblinks", "true");
 
 	// Altered settings for Android
-#ifdef __ANDROID__
-	settings->setDefault("screen_w", "0");
-	settings->setDefault("screen_h", "0");
 	settings->setDefault("performance_tradeoffs", "true");
 	settings->setDefault("max_simultaneous_block_sends_per_client", "20");
 	settings->setDefault("emergequeue_limit_diskonly", "32");
@@ -583,6 +580,9 @@ void set_default_settings()
 	settings->setDefault("post_processing_texture_bits", "8");
 	// We don't have working certificate verification...
 	settings->setDefault("curl_verify_cert", "false");
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
+	settings->setDefault("screen_w", "0");
+	settings->setDefault("screen_h", "0");
 
 	// Apply settings according to screen size
 	float x_inches = (float) porting::getDisplaySize().X /
--- a/src/filesys.cpp
+++ b/src/filesys.cpp
@@ -343,7 +343,7 @@ bool DeleteSingleFileOrEmptyDirectory(const std::string &path, bool log_error)
 
 std::string TempPath()
 {
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	return porting::path_cache;
 #else
 	const char *env_tmpdir = getenv("TMPDIR");
--- a/src/gui/guiEngine.cpp
+++ b/src/gui/guiEngine.cpp
@@ -400,7 +400,7 @@ void GUIEngine::run()
 		sound_volume_control(m_sound_manager.get(), device->isWindowActive());
 		m_sound_manager->step(dtime);
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 		m_menu->getAndroidUIInput();
 #endif
 	}
--- a/src/gui/guiFormSpecMenu.cpp
+++ b/src/gui/guiFormSpecMenu.cpp
@@ -3616,7 +3616,7 @@ void GUIFormSpecMenu::legacySortElements(std::list<IGUIElement *>::iterator from
 	reorderChildren(from, to, elements);
 }
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 void GUIFormSpecMenu::getAndroidUIInput()
 {
 	porting::AndroidDialogState dialogState = getAndroidUIInputState();
--- a/src/gui/guiFormSpecMenu.h
+++ b/src/gui/guiFormSpecMenu.h
@@ -311,7 +311,7 @@ public:
 	// This will only return a meaningful value if called after drawMenu().
 	core::rect<s32> getAbsoluteRect();
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	void getAndroidUIInput() override;
 #endif
 
--- a/src/gui/guiOpenURL.cpp
+++ b/src/gui/guiOpenURL.cpp
@@ -165,7 +165,7 @@ void GUIOpenURLMenu::drawMenu()
 	driver->draw2DRectangle(bgcolor, AbsoluteRect, &AbsoluteClippingRect);
 
 	gui::IGUIElement::draw();
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	getAndroidUIInput();
 #endif
 }
--- a/src/gui/guiPasswordChange.cpp
+++ b/src/gui/guiPasswordChange.cpp
@@ -153,7 +153,7 @@ void GUIPasswordChange::drawMenu()
 	driver->draw2DRectangle(bgcolor, AbsoluteRect, &AbsoluteClippingRect);
 
 	gui::IGUIElement::draw();
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	getAndroidUIInput();
 #endif
 }
@@ -249,7 +249,7 @@ std::string GUIPasswordChange::getNameByID(s32 id)
 	return "";
 }
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 void GUIPasswordChange::getAndroidUIInput()
 {
 	porting::AndroidDialogState dialogState = getAndroidUIInputState();
--- a/src/gui/guiPasswordChange.h
+++ b/src/gui/guiPasswordChange.h
@@ -43,7 +43,7 @@ public:
 	bool processInput();
 
 	bool OnEvent(const SEvent &event);
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	void getAndroidUIInput();
 #endif
 
--- a/src/gui/modalMenu.cpp
+++ b/src/gui/modalMenu.cpp
@@ -229,7 +229,7 @@ void GUIModalMenu::leave()
 
 bool GUIModalMenu::preprocessEvent(const SEvent &event)
 {
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	// display software keyboard when clicking edit boxes
 	if (event.EventType == EET_MOUSE_INPUT_EVENT &&
 			((event.MouseInput.Event == EMIE_LMOUSE_PRESSED_DOWN &&
@@ -359,7 +359,7 @@ bool GUIModalMenu::preprocessEvent(const SEvent &event)
 	return false;
 }
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 porting::AndroidDialogState GUIModalMenu::getAndroidUIInputState()
 {
 	// No dialog is shown
--- a/src/gui/modalMenu.h
+++ b/src/gui/modalMenu.h
@@ -8,7 +8,7 @@
 #include "irrlichttypes_bloated.h"
 #include "irr_ptr.h"
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	#include <porting_android.h>
 #endif
 
@@ -52,7 +52,7 @@ public:
 	virtual bool preprocessEvent(const SEvent &event);
 	virtual bool OnEvent(const SEvent &event) { return false; };
 	virtual bool pausesGame() { return false; } // Used for pause menu
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	virtual void getAndroidUIInput() {};
 	porting::AndroidDialogState getAndroidUIInputState();
 #endif
@@ -69,7 +69,7 @@ protected:
 
 	v2u32 m_screensize_old;
 	float m_gui_scale;
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	std::string m_jni_field_name;
 #endif
 
--- a/src/log.cpp
+++ b/src/log.cpp
@@ -10,7 +10,7 @@
 #include "exceptions.h"
 #include "filesys.h"
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 #include <android/log.h>
 #endif
 
@@ -55,7 +55,7 @@ private:
 
 Logger g_logger;
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 AndroidLogOutput stdout_output;
 AndroidLogOutput stderr_output;
 #else
@@ -84,7 +84,7 @@ thread_local LogStream derr_con(verbose_target);
 thread_local LogStream dout_con(trace_target);
 
 // Android
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 
 constexpr static unsigned int g_level_to_android[] = {
 	ANDROID_LOG_INFO,     // LL_NONE
--- a/src/log_internal.h
+++ b/src/log_internal.h
@@ -192,14 +192,14 @@ private:
 };
 
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 class AndroidLogOutput : public ICombinedLogOutput {
 public:
 	void logRaw(LogLevel lev, std::string_view line);
 };
 #endif
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 extern AndroidLogOutput stdout_output;
 extern AndroidLogOutput stderr_output;
 #else
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -617,7 +617,7 @@ static bool create_userdata_path()
 {
 	bool success;
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	if (!fs::PathExists(porting::path_user)) {
 		success = fs::CreateDir(porting::path_user);
 	} else {
@@ -670,7 +670,7 @@ namespace {
 
 static bool use_debugger(int argc, char *argv[])
 {
-#if defined(__ANDROID__)
+#if defined(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 	return false;
 #else
 #ifdef _WIN32
--- a/src/porting.cpp
+++ b/src/porting.cpp
@@ -25,7 +25,7 @@
 #if !defined(_WIN32)
 	#include <unistd.h>
 	#include <sys/utsname.h>
-	#if !defined(__ANDROID__)
+	#if !defined(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 		#include <spawn.h>
 	#endif
 #endif
@@ -33,7 +33,7 @@
 	#define _PSTAT64
 	#include <sys/pstat.h>
 #endif
-#if defined(__ANDROID__)
+#if defined(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 	#include "porting_android.h"
 	#include <android/api-level.h>
 #endif
@@ -281,7 +281,7 @@ static std::string detectSystemInfo()
 	delete[] lpVersionInfo;
 
 	return oss.str();
-#elif defined(__ANDROID__)
+#elif defined(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 	std::ostringstream oss;
 	struct utsname osinfo;
 	uname(&osinfo);
@@ -529,7 +529,7 @@ bool setSystemPaths()
 
 //// Android
 
-#elif defined(__ANDROID__)
+#elif defined(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 
 extern bool setSystemPaths(); // defined in porting_android.cpp
 
@@ -728,7 +728,7 @@ void initializePaths()
 	if (!setSystemPaths())
 		errorstream << "Failed to get one or more system-wide path" << std::endl;
 
-#  ifdef __ANDROID__
+#  ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	sanity_check(!path_cache.empty());
 #  elif defined(_WIN32)
 	path_cache = path_user + DIR_DELIM "cache";
@@ -825,7 +825,7 @@ bool secure_rand_fill_buf(void *buf, size_t len)
 
 #endif
 
-#ifndef __ANDROID__
+#ifndef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 
 void osSpecificInit()
 {
@@ -924,7 +924,7 @@ int mt_snprintf(char *buf, const size_t buf_size, const char *fmt, ...)
 	return c;
 }
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 // defined in porting_android.cpp
 extern void openURIAndroid(const char *url);
 #endif
@@ -939,7 +939,7 @@ static bool open_uri(const std::string &uri)
 
 #if defined(_WIN32)
 	return (intptr_t)ShellExecuteA(NULL, NULL, uri.c_str(), NULL, NULL, SW_SHOWNORMAL) > 32;
-#elif defined(__ANDROID__)
+#elif defined(DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK)
 	openURIAndroid(uri.c_str());
 	return true;
 #elif defined(__APPLE__)
--- a/src/porting.h
+++ b/src/porting.h
@@ -351,6 +351,6 @@ bool open_directory(const std::string &path);
 
 } // namespace porting
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 #include "porting_android.h"
 #endif
--- a/src/porting_android.cpp
+++ b/src/porting_android.cpp
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: LGPL-2.1-or-later
 // Copyright (C) 2014 celeron55, Perttu Ahola <celeron55@gmail.com>
 
-#ifndef __ANDROID__
+#ifndef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 #error This file may only be compiled for android!
 #endif
 
--- a/src/porting_android.h
+++ b/src/porting_android.h
@@ -4,7 +4,7 @@
 
 #pragma once
 
-#ifndef __ANDROID__
+#ifndef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 #error This header has to be included on Android port only!
 #endif
 
--- a/src/script/lua_api/l_mainmenu.cpp
+++ b/src/script/lua_api/l_mainmenu.cpp
@@ -1082,7 +1082,7 @@ int ModApiMainMenu::l_share_file(lua_State *L)
 
 	CHECK_SECURE_PATH(L, path, false)
 
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK
 	porting::shareFileAndroid(path);
 	lua_pushboolean(L, true);
 #else
--- a/src/server.cpp
+++ b/src/server.cpp
@@ -3335,7 +3335,7 @@ std::wstring Server::handleChat(const std::string &name,
 			Workaround for fixing chat on Android. Lua doesn't handle
 			the Cyrillic alphabet and some characters on older Android devices
 		*/
-#ifdef __ANDROID__
+#ifdef DISABLING_THIS_BECAUSE_IT_IS_FOR_ANDROID_4
 		line += L"<" + utf8_to_wide(name) + L"> " + wmessage;
 #else
 		line += utf8_to_wide(m_script->formatChatMessage(name, message));
